### PR TITLE
Tekton architecture blog post

### LINKED ISSUE #32: Blog:  Part 2 in series on Tekton, learn about Kabanero's DevOps Toolchain: Tekton Architecture
Blog Title: "Cloud-Native Devops: Tekton Architecture and Design"

Draft Blog by Mark Nuttall, is located at this PR: https://github.com/kabanero-io/blogs/pull/31

### PR DESCRIPTION
**Pull request by mnuttall**
Hello, 
This PR is for a blog post on "Cloud-Native Devops: Tekton Architecture and Design". I look forward to working through any review feedback. Many thanks in advance!

Regards, 
Mark

closes #32

### CODE REVIEW

File: drafts/2019-11-11-tekton-architecture.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+author_github: https://github.com/mnuttall
+seo-title: Title - Kabanero.io
+seo-description: It's all about the architecture.
+blog_description: "It's all about the architecture"
+---
+= Tekton Architecture
+Mark Nuttall <https://github.com/mnuttall>
+
+== Introduction
+This blog post is part of a series about Tekton, the cloud-native CI/CD engine that underpins Kabanero. It leads on from link:/publish/2019-11-05-Tekton-Enterprise-CI-CD.adoc[our previous article], ‘How Tekton can power your Enterprise’s continuous integration & continuous delivery.’ We go into more detail on how Tekton works, look at how it can be integrated into an end-to-end DevOps process and finally look forward to the following articles that will flesh out its use in an enterprise context.
```
Comment (anneqm) [inline drafts/2019-11-11-tekton-architecture.adoc]: Suggest changing URL to point to https://kabanero.io/blog/2019/11/05/Tekton-Enterprise-CI-CD.html on the published kabanero website/blogs page

File: drafts/2019-11-11-tekton-architecture.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+== Building a DevOps Pipeline
+Tekton gives us the pieces that we need to build a useful DevOps pipeline. Figure 3 below shows the high level flows involved. We show a very simple Pipeline without all the linting, testing and image scanning involved in a real world implementation. 
+
+[#img:basic-devops]
+.High level overview of a simple DevOps pipeline
+image::/img/basic-devops.png[High level overview of a DevOps pipeline]
+
+In this example, Pipelines are initiated as a result of Git activity by a developer. See Weaveworks' excellent https://www.weave.works/blog/what-is-gitops-really[What Is GitOps Really?] article for more on why this model is increasingly popular. Figure 3 shows a 'webhook' initiating a Tekton Pipeline. Webhooks are outbound HTTP POST messages with a payload specific to the particular event provider provider. These payloads contain most of the information necessary to kick off a Tekton PipelineRun. We discuss webhooks in a bit more detail below.
```
Comment (anneqm) [inline drafts/2019-11-11-tekton-architecture.adoc]: Is duplicate provider provider intentional?

**Review by dewan-ahmed — COMMENTED** [review]
1. "the cloud-native CI/CD engine" - Would it be better to word Kubernetes-native CI/CD engine (as Tekton has it on their website)?
2. "Tekton’s basic building block is called a Task." --> Isn't it a step? Multiple steps form a Task
3. Under the Building Blocks section, would it make sense to add a note on parallel tasks that don't require other tasks to finish?
4. "Tekton is a ‘cloud native’ build system" --> We can deploy using Tekton as well.
5. New line required after Figure2 Caption
6. New line required after Figure3 Caption
7. "Figure three raises some further questions:" --> Figure3

**Comment by mnuttall** [comment]
@anneqm resolved thank you. 
@dewan-ahmed the lack of new lines is due to the .adoc format being rubbish. I spoke to @alohr51 about them; he was going to clean them up before @nateziemann commented on https://github.com/kabanero-io/blogs/issues/32. Otherwise thank you I've integrated your suggestions.

**Comment by anneqm** [comment]
This blog will be published via this issue: https://github.ibm.com/IBMCode/IBMCodeContent/issues/3578 on developer.ibm.com. Will update again with final link when it's published.

**Comment by anneqm** [comment]
This blog has now moved to publishing stage on developer.ibm.com. Here's the related tracking issue: https://github.ibm.com/IBMCode/Code-Articles/issues/4545

**Pull request closed without merging**